### ISSUE TITLE
Construction Script

### ISSUE THREAD

**Issue by dantarion** [ncsoft/Unreal.js#175]
I would like to run a Unreal.JS script in the Construction Script of my Actor.

My actor has a Javascript component, and it spawns a procredual map by populating InstancedStaticMeshComponents. However, it doesn't run until BeginPlay.

This means I cannot build lighting. My idea so far is as follows.

1. Have Unreal.JS-based script spawn procredual map on Construction
2. Hand place additional actors around procedural map.
3. Build Lighting....

I have also thought about having my map generation take the form of an Editor extension, but I don't know enough about UE4 yet to do that part.

**Anyways, How can I load and run a JS script as part of the Actor Construction Script?**
If it isn't possible, what is the best way to populate a map inside the editor using Unreal.JS?

**Comment by crocuis** [comment]
If it is not a runtime generated mesh, it would be better to spawn the generated mesh by configuring the Editor Extension with Javascript. If it should be created at runtime, even if Javascript is called in the Construnction Script, it does not seem to be necessary to build Light if a mesh is created at runtime.

**Comment by dantarion** [comment]
I am not aiming to do it at runtime, I am aiming it to do it in my construction script. However, I figured out another way to do it, ill update this issue just to explain it for others.

I wanted a maze/building generator that I could place on my map, tweak parameters, click  "Regenerate" and have it populate a blueprint actors InstancedStaticMeshComponents.

I thought I needed to do this in the construction script, so I could generate the map while it is placed on the map. Instead, I created a function that is callable in the editor, and now I have a button I can click in the Actor's properties that does the regeneration. As a result, the InstancedStaticMeshComponents remain populated and the generation does NOT have to be done on launch, the procedural content is basically baked into the map.

The next step is converting it to an editor extention an trying to see if I can completely remove the runtime dependency on Unreal.JS. 

Thanks for the response it helped me find my own solution :D
